dataland-frontend/src/components/forms/CreateSFDRDataset.vue

```diff
@@ -0,0 +1,330 @@
+import DateFormField from "@/components/forms/parts/fields/DateFormField.vue";
+import SingleSelectFormField from "@/components/forms/parts/fields/SingleSelectFormField.vue";
+import MultiSelectFormField from "@/components/forms/parts/fields/MultiSelectFormField.vue";
+import AddressFormField from "@/components/forms/parts/fields/AddressFormField.vue";
+import RadioButtonsFormField from "@/components/forms/parts/fields/RadioButtonsFormField.vue";
+import SubmitButton from "@/components/forms/parts/SubmitButton.vue";
+import SubmitSideBar from "@/components/forms/parts/SubmitSideBar.vue";
+import YesNoNaFormField from "@/components/forms/parts/fields/YesNoNaFormField.vue";
+import UploadReports from "@/components/forms/parts/UploadReports.vue";
+import DataPointFormField from "@/components/forms/parts/kpiSelection/DataPointFormField.vue";
+import PercentageFormField from "@/components/forms/parts/fields/PercentageFormField.vue";
+import ProductionSitesFormField from "@/components/forms/parts/fields/ProductionSitesFormField.vue";
+import { objectDropNull, ObjectType } from "@/utils/UpdateObjectUtils";
+import { smoothScroll } from "@/utils/SmoothScroll";
+import { DocumentToUpload, ReportToUpload, uploadFiles } from "@/utils/FileUploadUtils";
+import MostImportantProductsFormField from "@/components/forms/parts/fields/MostImportantProductsFormField.vue";
+import { Subcategory } from "@/utils/GenericFrameworkTypes";
+import ProcurementCategoriesFormField from "@/components/forms/parts/fields/ProcurementCategoriesFormField.vue";
+
+export default defineComponent({
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+    };
+  },
+  name: "CreateSfdrDataset",
+  components: {
+    SubmitButton,
+    SubmitSideBar,
+    UploadFormHeader,
+    SuccessMessage,
+    FailMessage,
+    FormKit,
+    Card,
+    PrimeButton,
+    Calendar,
+    YesNoFormField,
+    InputTextFormField,
+    FreeTextFormField,
+    NumberFormField,
+    DataPointFormField,
+    DateFormField,
+    SingleSelectFormField,
+    MultiSelectFormField,
+    NaceCodeFormField,
+    AddressFormField,
+    RadioButtonsFormField,
+    YesNoNaFormField,
+    PercentageFormField,
+    ProductionSitesFormField,
+    MostImportantProductsFormField,
+    ProcurementCategoriesFormField,
+    UploadReports,
+  },
+  directives: {
+    tooltip: Tooltip,
+  },
+  emits: ["datasetCreated"],
+  data() {
+    return {
+      formId: "createSFDRForm",
+      waitingForData: true,
+      dataDate: undefined as Date | undefined,
+      companyAssociatedSfdrData: {} as CompanyAssociatedDataSfdrData,
+      sfdrDataModel: sfdrDataModel,
+      route: useRoute(),
+      message: "",
+      smoothScroll: smoothScroll,
+      uploadSucceded: false,
+      postSfdrDataProcessed: false,
+      messageCounter: 0,
+      checkCustomInputs,
+      documents: [] as DocumentToUpload[],
+      referencedReportsForPrefill: {},
```

Comment:
Checking ...

Reply:
Done.

Reply:
I think technically it is more or less the same, but let's type this as:

`Map<String, CompanyReport> | undefined`, because then it is literally the same as in the respective model `SfdrGeneralGeneral.kt` for the field `referencedReports`.
Furthermore: This is currently initialized as empty object `{}`. Is this necessary?  Since you typed it in a way, that it can be `undefined`, we could also remove this inital-empty-object-setting and just let it be undefined at the beginning.

Or will this break anything?

Reply:
I changed it to use Map(). Apart from minor differences, it is pretty much the same as `CreateLksgDataset.vue`

`documents: new Map() as Map<string, DocumentToUpload>`

Reply:
So is my suggestion ok for you?

Reply:
From what I can see in the code, it has already been updated with this type 👍

Reply:
Yes has been typed now. But different than in my suggestion.

And my suggestion to set it to `undefined` at the beginning has not been tried out / responded to / addressed yet.
Do you think this is possible?

Reply:
It seems this has been forgotten. => friendly reminder

Reply:
@MarcinStanczyk7 started to have look at this 10 minutes ago :)

Reply:
I removed undefined from the type instead we check if the variable exists while we assignment new value

Reply:
Ok then let's stick to it.
But why has it the type "undefined" too ?